src/frontend/src/optimizer/rule/logical_filter_expression_simplify_rule.rs

```diff
@@ -0,0 +1,252 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use fixedbitset::FixedBitSet;
+use risingwave_common::types::ScalarImpl;
+use risingwave_connector::source::DataType;
+
+use crate::expr::{Expr, ExprImpl, ExprRewriter, ExprType, FunctionCall};
+use crate::optimizer::plan_expr_visitor::strong::Strong;
+use crate::optimizer::plan_node::{ExprRewritable, LogicalFilter, PlanTreeNodeUnary};
+use crate::optimizer::rule::{BoxedRule, Rule};
+use crate::optimizer::PlanRef;
+
+/// Specially for the predicate under `LogicalFilter`
+pub struct LogicalFilterExpressionSimplifyRule {}
+impl Rule for LogicalFilterExpressionSimplifyRule {
+    /// The pattern we aim to optimize, e.g.,
+    /// 1. (NOT (e)) OR (e) => True
+    /// 2. (NOT (e)) AND (e) => False
+    /// NOTE: `e` should only contain at most a single column
+    /// otherwise we will not conduct the optimization
+    fn apply(&self, plan: PlanRef) -> Option<PlanRef> {
+        let filter: &LogicalFilter = plan.as_logical_filter()?;
+        let mut rewriter = ExpressionSimplifyRewriter {};
+        let input = filter.input().rewrite_exprs(&mut rewriter);
+        Some(filter.clone_with_input(input).into())
```

Comment:
I think we can still match the previous pattern `LogicalFilter->LogicalShare->LogicalFilter` to avoid the additional overhead for all filters.

Reply:
Make sense, let me change it back.